### PR TITLE
feat(parser): Relax the order requirement about the COPY options

### PR DESCRIPTION
**Pull request by flaneur2020**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Currently the options to the COPY statement requires order as:

```
            ~ ( FILES ~ "=" ~ "(" ~ #comma_separated_list0(literal_string) ~ ")")?
            ~ ( PATTERN ~ "=" ~ #literal_string)?
            ~ ( FILE_FORMAT ~ "=" ~ #options)?
            ~ ( VALIDATION_MODE ~ "=" ~ #literal_string)?
            ~ ( SIZE_LIMIT ~ "=" ~ #literal_u64)?
            ~ ( MAX_FILE_SIZE ~ "=" ~ #literal_u64)?
            ~ ( SPLIT_SIZE ~ "=" ~ #literal_u64)?
            ~ ( SINGLE ~ "=" ~ #literal_bool)?
            ~ ( PURGE ~ "=" ~ #literal_bool)?
            ~ ( FORCE ~ "=" ~ #literal_bool)?
```

If we put FORCE option in front of PURGE, we'll get a syntax error. The args order is hard to memorize, makes it hard for users to use the COPY statment.

This PR relax the order requirement about the COPY options, so we can put PURGE option in front of the SINGLE option in our SQL.

### CODE REVIEW

File: src/query/ast/src/ast/statements/copy.rs [hunk 1]
```diff
@@ -189,3 +189,33 @@ impl Display for UriLocation {
         Ok(())
     }
 }
+
+pub enum CopyOptionItem {
```
Comment (Xuanwo) [inline src/query/ast/src/ast/statements/copy.rs:193]: Clever!

Comment (BohuTANG) [reply src/query/ast/src/ast/statements/copy.rs:193]: 🚀

File: src/query/ast/src/ast/statements/copy.rs [hunk 2]
```diff
@@ -189,3 +189,33 @@ impl Display for UriLocation {
...
+    ValidationMode(String),
+    SizeLimit(usize),
+    MaxFileSize(usize),
+    SplitSize(usize),
+    Single(bool),
+    Purge(bool),
+    Force(bool),
+}
+
+impl CopyOptionItem {
```
Comment (Xuanwo) [inline src/query/ast/src/ast/statements/copy.rs:206]: How about using `CopyOption` for simple?

File: src/query/ast/src/ast/statements/copy.rs [hunk 3]
```diff
@@ -189,3 +189,33 @@ impl Display for UriLocation {
...
+    pub fn apply(&self, stmt: &mut CopyStmt<'_>) {
```
Comment (Xuanwo) [inline src/query/ast/src/ast/statements/copy.rs:207]: How about impl CopyStmt with `pub fn apply_option(&mut self, opt: CopyOption)`? We can move and consume the CopyOption directly without an extra clone.

**Comment by Xuanwo** [comment]
Cc @andylokandy for reviewing.

File: src/query/ast/src/parser/statement.rs [hunk 4]
```diff
@@ -750,53 +750,33 @@ pub fn statement(i: Input) -> IResult<StatementMsg> {
...
-                file_format: file_format.map(|v| v.2).unwrap_or_default(),
-                validation_mode: validation_mode.map(|v| v.2).unwrap_or_default(),
-                size_limit: size_limit.map(|v| v.2).unwrap_or_default() as usize,
-                max_file_size: max_file_size.map(|v| v.2).unwrap_or_default() as usize,
-                split_size: split_size.map(|v| v.2).unwrap_or_default() as usize,
-                single: single.map(|v| v.2).unwrap_or_default(),
-                purge: purge.map(|v| v.2).unwrap_or_default(),
-                force: force.map(|v| v.2).unwrap_or_default(),
-            })
+                files: Default::default(),
```
Comment (andylokandy) [inline src/query/ast/src/parser/statement.rs:765]:
```suggestion
                .. Default::default(),
```

**Review by andylokandy — APPROVED** [review]

File: src/query/ast/src/parser/statement.rs (hunk 4 above)
Comment (flaneur2020) [reply src/query/ast/src/parser/statement.rs]: TIL quq

File: src/query/ast/src/ast/statements/copy.rs (hunk 3 above)
Comment (flaneur2020) [reply src/query/ast/src/ast/statements/copy.rs]: yep, lemme change it to move

**Pull request merged**